cloud/aws/actuators/machine/actuator.go

```diff
@@ -66,42 +69,47 @@ type ActuatorParams struct {
 
 // NewActuator returns an actuator.
 func NewActuator(params ActuatorParams) (*Actuator, error) {
+	e, ok := params.EC2Service.(*ec2svc.Service)
+	if !ok {
+		return nil, errors.New("EC2Service is not a real ec2service")
+	}
 	return &Actuator{
 		codec:          params.Codec,
 		ec2:            params.EC2Service,
 		machinesGetter: params.MachinesGetter,
+		infra: &infra.Infra{
```

Comment:
This is an experiment. I am playing around with a new idea. any feedback welcome

Reply:
Is this an attempt to separate out the business logic from the AWS wrapper?

Reply:
it is. What do you think?

Reply:
I like the approach in general, just a matter of getting the naming and code structure to be more intuitive :)